### PR TITLE
Search js

### PR DESCRIPTION
**Pull request by ZhyrovDmytro**

### CODE REVIEW

File: src/app/components/HistoryList.js [hunk 1]
```diff
@@ -1,6 +1,7 @@
 import React from 'react';
+import PropTypes from 'prop-types';
 
-export default (props) => {
+export default function historyList(props) {
```
Comment (fidelman) [inline src/app/components/HistoryList.js]: Could you revert to arrow function?

File: src/app/components/HistoryList.js [hunk 2]
```diff
@@ -25,4 +26,9 @@ export default (props) => {
             </ul>
         </div>
     );
+}
+
+historyList.propTypes = {
+    suggestItemToShow: PropTypes.array.isRequired,
```
Comment (fidelman) [inline src/app/components/HistoryList.js]: Could you describe the array?
The array of what? in details please

File: src/app/components/HistoryList.js (hunk 1 above)
Comment (fidelman) [reply src/app/components/HistoryList.js]: All comments are related to all files

File: src/app/components/InfoOptions.js [hunk 3]
```diff
@@ -23,4 +27,14 @@ export default (props) => {
             </a>
         </div>
     );
+}
+
+infoOptions.propTypes = {
+    info: PropTypes.PropTypes.shape({
```
Comment (fidelman) [inline src/app/components/InfoOptions.js]: 👍

File: src/app/components/ItemInfo.js [hunk 4]
```diff
@@ -22,3 +26,14 @@ export default function ItemInfo(props) {
...
     );
 }
+
+ItemInfo.propTypes = {
+    info: PropTypes.PropTypes.shape({
+        links: PropTypes.PropTypes.shape({
+            html: PropTypes.string.isRequired
+        }).isRequired,
+        user: PropTypes.PropTypes.shape({
+            last_name: PropTypes.string
```
Comment (fidelman) [inline src/app/components/ItemInfo.js:36]: camelCase?

File: src/app/components/ModalWindow.js [hunk 5]
```diff
@@ -33,11 +37,11 @@ export default class ModalWindow extends Component {
                     <div className="modal__item">
                         <div className="modal__profile">
                             <div className="modal__info--profile">
-                                <picture>
+                                <div>
```
Comment (fidelman) [inline src/app/components/ModalWindow.js:40]: 👍

File: src/app/components/ModalWindow.js [hunk 6]
```diff
@@ -75,3 +79,19 @@ export default class ModalWindow extends Component {
...
 }
+
+ModalWindow.propTypes = {
+    image: PropTypes.PropTypes.shape({
+        user: PropTypes.PropTypes.shape({
+            links: PropTypes.PropTypes.shape({
+                html: PropTypes.string.isRequired
+            }).isRequired,
+            username: PropTypes.string.isRequired,
+            location: PropTypes.string
```
Comment (fidelman) [inline src/app/components/ModalWindow.js:90]: 👍

File: src/app/components/ResultList.js [hunk 7]
```diff
@@ -1,13 +1,16 @@
...
 
-export default (props) => {
+export default function resultList(props) {
 
     const createItem = props.images.map((image) => {
         return (
             <figure
-                key={`_${image.id}`}
-                // key={`_${image.id}${Math.random().toString(36).slice(-5)}`}
+                key={`_${image.id}${Math.random().toString(36).slice(-5)}`}
```
Comment (fidelman) [inline src/app/components/ResultList.js]: Create a custom method to generate a random ID as a helper or use libraries like `uuid`

File: src/app/components/Search.js [hunk 8]
```diff
@@ -100,14 +97,26 @@ export default class Search extends Component {
         this.setState({ loadNextPage: true });
     }
 
-    loaderActive = () => {
+    activateLoadre = () => {
```
Comment (fidelman) [inline src/app/components/Search.js]: Loadre :)

Are you sure you ONLY active a loader in this function? :)

File: src/app/components/Search.js [hunk 9]
```diff
@@ -100,14 +97,26 @@ export default class Search extends Component {
...
         this.setState({
             loaderIsActive: !this.state.loaderIsActive
         });
     }
 
     render() {
         const { images } = this.state;
+        const loader = this.state.loaderIsActive === true && <Loader />;
+        const noImages = !this.state.noImages &&
```
Comment (fidelman) [inline src/app/components/Search.js:109]: wrap in brackets the view please

File: src/app/components/base/Button.js [hunk 10]
```diff
@@ -1,14 +1,20 @@
...
+import PropTypes from 'prop-types';
 
 export default function button(props) {
     const buttonClass = props.className;
     const content = props.content;
     return (
         <button
             className={buttonClass}
-            onClick={(event) => this.handleClick(event)}
+            onClick={() => this.handleClick()}
```
Comment (fidelman) [inline src/app/components/base/Button.js]: have you tested after changes?
I am not sure it will work

Comment (fidelman) [reply src/app/components/base/Button.js]: At least, an event is not passed

**Review by fidelman — APPROVED** [review]
Very very nice, polish a little bit more

File: src/app/components/InfoOptions.js (hunk 3 above)
Comment (fidelman) [reply src/app/components/InfoOptions.js]: cool

**Review by fidelman — CHANGES_REQUESTED** [review]

File: src/app/components/ItemInfo.js (hunk 4 above)
Comment (ZhyrovDmytro) [reply src/app/components/ItemInfo.js:36]: I receive object with last_name

File: src/app/components/base/Button.js (hunk 10 above)
Comment (ZhyrovDmytro) [reply src/app/components/base/Button.js]: I found that I didn't use <Button /> at all =)

But we decided that i don't need to pass event on our previous meeting?

**Review by fidelman — APPROVED** [review]

**Pull request merged**